test/end-to-end/test/test_createRecipe.js

```diff
@@ -22,15 +20,14 @@ describe('Create Recipe Page', () => {
   const createRecipePage = new CreateRecipePage(pageConfig.recipe.simple.name
     , pageConfig.recipe.simple.description);
 
-// Switch to welder-web iframe if welder-web is integrated with Cockpit.
-// Cockpit web service is listening on TCP port 9090.
   beforeEach(() => {
-    helper.gotoURL(nightmare = new Nightmare(), recipesPage);
+    helper.gotoURL(nightmare = new Nightmare(pageConfig.nightmareTimeout), recipesPage);
   });
 
   describe('Input Data Validation Test', () => {
-    describe('Required Field Missing #acceptance', () => {
-      test('should show alert message by clicking Save button when create recipe without name @create-recipe-page', (done) => {
+    describe('Required Field Missing', () => {
+      const testSpec1 = test('should show alert message by clicking Save button when create recipe without name',
+      (done) => {
```

Comment:
Can you explain why remove the # and @ tagging from test case names? 

Also why do test specs need to be defined as constants and assigned to variable names?

Reply:
Two reasons of removing @ and #:
1. The file name of screenshot comes from the full name of test case (describe+test). Some of test cases has long test name, which cause a very long filename.
2. the # and @ should not be in filename, sometimes # and @ will cause problem.

To avoid these issues, I removed # and @. But the function is still working with test file name and describe name. That means if I'd like to run edit recipe page tests only, just append "-- editRecipe.js" at the end of npm run test.

The const variable is to get full test name which will be used as screenshot file name.

Reply:
> The const variable is to get full test name which will be used as screenshot file name.

isn't there something like `this` awailable inside of the test spec ? From what I can tell the spec variable is passed as parameter to the helper.gotoError function.

Reply:
`this` does not work here, because the test object has to be returned by `test()`or `it()` function and there's no object variable referencing to test object. So I have to create a variable to get test object to get full name.